shell/platform/android/io/flutter/embedding/engine/android/AndroidTouchProcessor.java

```diff
@@ -114,14 +129,45 @@ public boolean onTouchEvent(MotionEvent event) {
     }
 
     // Verify that the packet is the expected size.
-    assert packet.position() % (POINTER_DATA_FIELD_COUNT * BYTES_PER_FIELD) == 0;
+    if (packet.position() % (POINTER_DATA_FIELD_COUNT * BYTES_PER_FIELD) != 0) {
+      throw new AssertionError("Packet position is not on field boundary");
+    }
 
     // Send the packet to Flutter.
     renderer.dispatchPointerDataPacket(packet, packet.position());
 
     return true;
   }
 
+  /**
+   * Sends the given generic {@link MotionEvent} data to Flutter in a format that Flutter
+   * understands.
+   *
+   * Generic motion events include joystick movement, mouse hover, track pad touches, scroll wheel
+   * movements, etc.
+   */
+  public boolean onGenericMotionEvent(MotionEvent event) {
+    // Method isFromSource is only available in API 18+ (Jelly Bean MR2)
+    // Mouse hover support is not implemented for API < 18.
+    boolean isPointerEvent = Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2
+        && event.isFromSource(InputDevice.SOURCE_CLASS_POINTER);
+    if (!isPointerEvent || event.getActionMasked() != MotionEvent.ACTION_HOVER_MOVE) {
```

Comment:
Scroll wheel support just landed; I don't know if you want to go ahead and implement it now, or save it for a follow-up.

Reply:
Is there a PR?

Reply:
https://github.com/flutter/flutter/pull/22762

Reply:
@gspencergoog @stuartmorgan I don't see any Java changes in that PR.  Are there corresponding changes that need to take place in the embedding?

Reply:
That PR is the Flutter framework support for understanding scroll events; shells/embedders need to actually send the events for scroll wheels/trackpad scrolls to actually do anything.

When I landed the engine support for the new event fields I put minimal placeholders in the Android code to keep everything compiling, which is what you're moving in this PR (hard-coded `signalKind` of `PointerSignalKind.NONE` and `scroll_delta_x`/`scroll_delta_y` of `0.0`), but since you're in this code right now you could potentially add the actual event support instead of moving the dummy values, but it's up to you.

Reply:
I think it would be better to let you continue on your path.  The reason for this PR is just that new functionality was added in 1 place when it needs to be in 2 places, which will be true moving forward.  I just ask that you add me, @jason-simmons and @mklim to any future PRs in the Android embedding so that we can help ensure the correct integration with the new embedding architecture.

With regard to this PR, do all of my migrations look good?

Reply:
> I think it would be better to let you continue on your path. The reason for this PR is just that new functionality was added in 1 place when it needs to be in 2 places, which will be true moving forward.

To clarify: my path doesn't involve adding the Android shell code for generating scroll events. And given the deletion in this PR, it looks like that code would now only need to be implemented in this one class, unless I'm misunderstanding, which is why I thought you might want to add it now. But I can definitely understand not wanting to conflate the move with adding functionality.

> I just ask that you add me, @jason-simmons and @mklim to any future PRs in the Android embedding so that we can help ensure the correct integration with the new embedding architecture.

Will do for adding new signal types/data.

> With regard to this PR, do all of my migrations look good?

LGTM for the limited portions I'm familiar with.